src/api.ts

```diff
@@ -340,10 +340,21 @@ export function prosopoMiddleware(env): Router {
      * @param {string} dappId - Dapp Contract AccountId
      * @return {Captcha} - The Captcha data
      */
-    router.get('/v1/prosopo/provider/captcha', function (req, res, next) {
-        // query database for captcha
-        // send one solved, one unsolved
-        next();
+    router.get('/v1/prosopo/provider/captcha/:datasetId', async function (req, res, next) {
+        try {
+            const {datasetId} = req.params;
+            if (!datasetId) {
+                throw new BadRequest(ERRORS.API.PARAMETER_UNDEFINED.message);
+            }
+            //return one solved and one unsolved
+            const solved = await tasks.getCaptchaWithProof(datasetId, true, 1)
+            const unsolved = await tasks.getCaptchaWithProof(datasetId, false, 1)
+            let result = [solved[0], unsolved[0]];
```

Comment:
API-wise I can see some added clarity if we specify which is which, so like
`{
    "solved":{solved captcha json},
    "unsolved":{unsolved captcha json}
}`

Reply:
The person receiving the captchas is the `Dapp User` (e.g. a user of Uniswap or some other decentralised app). They are being asked to solve the captchas to prove they are human. Therefore, we are giving them an advantage if we tell them which is unsolved as they know they can submit anything back for that captcha - we don't know if they are correct or not.

We send one solved captcha and one unsolved captcha when a Provider wants data labelled. For example, they have a bunch of pictures of cars/buses/trains and they want to know which pictures are buses. Some of the captchas are already labelled (solved) but some are not labelled at all (unsolved). The unsolved ones will become solved once a threshold number of people have committed their answers.

Good point though - it would normally be clearer to separate the two! You have also made me realise that this feature requires a follow up issue: Allow Providers to send **only** solved captchas (when Providers are purely providing a captcha service and not concerned with data labelling).

Reply:
The order of solved and unsolved should probably also be randomised.